src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -272,15 +277,18 @@ public void exportIllustrations(String query, Integer startYear, Integer endYear
         }
         // Query Solr
         SolrQuery finalQuery = illustrationSolrQuery(query, startYear, endYear, max);
-        Stream<SolrDocument> docs = streamSolr(finalQuery, max);
+        Stream<SolrDocument> docs = streamSolr(finalQuery);
         // Create metadata file, that has to be added to output zip
         Map<String, Object> metadataMap = makeMetadataMap(query, startYear, endYear);
         // Create metadata objects
         HashSet<String> uniqueUUIDs = new HashSet<>();
-        Stream<IllustrationMetadata> illustrationMetadata = docs.flatMap(doc -> getMetadataForIllustrations(doc, uniqueUUIDs));
+        Stream<IllustrationMetadata> illustrationMetadata = docs.
+                flatMap(doc -> getMetadataForIllustrations(doc, uniqueUUIDs).
+                limit(max));
 
         // Streams illustration from URL to zip file with all illustrations
-        createZipOfImages(illustrationMetadata, output, metadataMap, exportFormat);
+        AtomicInteger count = createZipOfImages(illustrationMetadata, output, metadataMap, exportFormat);
+        log.info("Found: '" + count + "' unique UUIDs in query");
```

Comment:
Maybe "Exported" instead of "Found". Also, please state `query`, `startYear` and `endYear` as part of the log message.

Reply:
Fully optional how you want to do this, but note that instead of logging with
```
        log.info("Exported: '" + count + "' unique UUIDs from query: '" + query + "' with startYear: '" + startYear + "' and endYear: '" + endYear + "'");
```
you can use templating
```
        log.info("Exported: '{} unique UUIDs from query: '{}' with startYear: {} and endYear: {}", count, query, startYear, endYear);
```